### DISCUSSION TITLE
How to differentiate between a disconnect and move?

### DISCUSSION THREAD

**Discussion by sda3-14** [discord-jda/JDA#1598] [Questions and Help] [answered]
I am trying to write a music bot, but I am unable to differentiate between someone disconnecting the bot and someone moving the bot. For both events, the connection status changes to `DISCONNECTED_KICKED_FROM_CHANNEL`. How do I check if it will reconnect to another channel or if it was kicked?

**Comment by DManstrator** [comment, marked as answer]
You can use the [`GuildVoiceMoveEvent`](https://ci.dv8tion.net/job/JDA/javadoc/net/dv8tion/jda/api/events/guild/voice/GuildVoiceMoveEvent.html) for that. It also mentions how a move is differentiated from a leave.

**Comment by sda3-14** [comment]
> You can use the [`GuildVoiceMoveEvent`](https://ci.dv8tion.net/job/JDA/javadoc/net/dv8tion/jda/api/events/guild/voice/GuildVoiceMoveEvent.html) for that. It also mentions how a move is differentiated from a leave.

Okay, thanks!
